ksp_plugin_adapter/flight_planner.cs

```diff
@@ -7,34 +7,254 @@
 namespace principia {
 namespace ksp_plugin_adapter {
 
-class FlightPlanner {
-  public FlightPlanner() {
-    burns_ = new List<BurnEditor>();
+class FlightPlanner : WindowRenderer {
+  public FlightPlanner(PrincipiaPluginAdapter adapter,
+                       IntPtr plugin) : base(adapter) {
+    adapter_ = adapter;
+    plugin_ = plugin;
+    window_rectangle_.x = UnityEngine.Screen.width / 2;
+    window_rectangle_.y = UnityEngine.Screen.height / 3;
+    final_time_ = new DifferentialSlider(
+                label : "Plan length",
+                unit : null,
+                log10_lower_rate : Log10TimeLowerRate,
+                log10_upper_rate : Log10TimeUpperRate,
+                min_value        : 10,
+                max_value        : double.PositiveInfinity,
+                formatter        : value =>
+                    FormatPositiveTimeSpan(
+                        TimeSpan.FromSeconds(
+                            value - plugin_.FlightPlanGetInitialTime(
+                                        vessel_.id.ToString()))));
+
+  }
+
+  protected override void RenderWindow() {
+    var old_skin = UnityEngine.GUI.skin;
+    UnityEngine.GUI.skin = null;
+    if (show_planner_) {
+      window_rectangle_ = UnityEngine.GUILayout.Window(
+                              id         : this.GetHashCode(),
+                              screenRect : window_rectangle_,
+                              func       : RenderPlanner,
+                              text       : "Flight plan",
+                              options    : UnityEngine.GUILayout.MinWidth(500));
+    }
+    UnityEngine.GUI.skin = old_skin;
+  }
+
+  public void RenderButton() {
+    var old_skin = UnityEngine.GUI.skin;
+    UnityEngine.GUI.skin = null;
+    if (UnityEngine.GUILayout.Button("Flight plan...")) {
+      show_planner_ = !show_planner_;
+    }
+    UnityEngine.GUI.skin = old_skin;
   }
 
-  public void Render() {
+  private void RenderPlanner(int window_id) {
```

Comment:
This function has 150.  Could we split it in smaller bits?  (Not too small so that everybody can see them.)  The innermost loop, in particular, looks like a good candidate.